server/api.go

```diff
@@ -208,6 +231,34 @@ func (a *apiServer) Publish(ctx context.Context, req *client.PublishRequest) (
 	return resp, err
 }
 
+func (a *apiServer) resumeStream(ctx context.Context, req *client.PublishRequest) error {
+	if req.Stream == "" {
+		return nil
+	}
+	stream := a.metadata.GetStream(req.Stream)
+	if stream == nil {
+		status.Error(codes.NotFound, fmt.Sprintf("No such stream: %s", req.Stream))
+	}
+	partition, ok := stream.partitions[0] // Only check the first partition for now
+	if !ok {
+		status.Error(codes.NotFound, fmt.Sprintf("No partition in stream: %s", req.Stream))
+	}
+	if partition.paused {
+		// This partition (and thus, this stream) is paused.
+		// We need to re-create the stream before being able to use it.
+		protoPart := partition.Partition
+		for i := 0; i < len(stream.partitions); i++ {
+			if e := a.metadata.CreatePartition(ctx, &proto.CreatePartitionOp{
+				Partition: protoPart,
+			}); e != nil {
+				a.logger.Errorf("api: Failed to resume stream partition %d: %v", i, e.Err())
+				e.Err()
```

Comment:
I think this is supposed to be a `return`?

Reply:
Should be fixed in 1bb7320.